lib/model/repositories/MockListRepository.dart

```diff
@@ -0,0 +1,87 @@
+import 'dart:async';
+
+import 'package:farmsmart_flutter/model/model/EntityCollectionInterface.dart';
+import 'package:farmsmart_flutter/model/repositories/BasicRepositoryInterface.dart';
+typedef String IdentifyEntity<T>(T value);
+
+class MockListRepository<T> implements BasicRepositoryInterface<T> {
```

Comment:
this same pattern will be present in the firebase version. so should be quick to add firebase repos.